### PR TITLE
[EPMRPP-42845] SAML, LDAP, AD single interface

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.io.Serializable;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class AbstractAuthResource implements Serializable {
+
+	private String type;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java]: should it be in integration params? do we want to have a separate resource for auth integrations?

Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java]: done

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.epam.ta.reportportal.ws.model.integration.auth;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java]: headers

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/SamlProvidersResource.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.epam.ta.reportportal.ws.model.integration.auth;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/SamlProvidersResource.java]: headers

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/UpdateAuthRQ.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.epam.ta.reportportal.ws.model.integration.auth;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/UpdateAuthRQ.java]: headers

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/AbstractAuthResource.java]: done

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/SamlProvidersResource.java (hunk 3 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/SamlProvidersResource.java]: done

File: src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/UpdateAuthRQ.java (hunk 4 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/integration/auth/UpdateAuthRQ.java]: done

**Pull request merged**